shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1311,15 +1311,14 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool)
     def goDecl(d: NuMember)(implicit ectx: ExpCtx): NuDecl = d match {
       case TypedNuAls(level, td, tparams, body) =>
         ectx(tparams) |> { implicit ectx =>
-          NuTypeDef(td.kind, td.nme, td.tparams, Tup(Nil), N, td.isPlain, S(go(body)), Nil, N, N, TypingUnit(Nil))(
+          NuTypeDef(td.kind, td.nme, td.tparams, S(Tup(Nil)), N, S(go(body)), Nil, N, N, TypingUnit(Nil))(
```

Comment:
```suggestion
          NuTypeDef(td.kind, td.nme, td.tparams, N, N, S(go(body)), Nil, N, N, TypingUnit(Nil))(
```

Reply:
Also it would be good to make sure type aliases were parsed into `NuTypeDef` instances with `N` as params.